module-2.md

```diff
@@ -0,0 +1,81 @@
+# 2: Setting up your first node 
+
+Total length:  1.5 hour
+Planned:      1.25 hour
+
+| **Section**                                        | **Length** |
+|----------------------------------------------------|------------|
+| **1: Topic introduction**                          | 15 min     |
+| **2: Hands-on and material exploration**           | 40 min     |
+| **3: Discussion and reflection activity**          | 20 min     |
+| **4: Wrap-up and week ahead**                      | 5 min      |
+
+## 1: Topic introduction
+
+- Summarize "homework" takeaways and answer questions around basic networking
+- Discuss network security and privacy primitives
+
+## 2: Hands-on and material exploration
+
+### Objectives
+
+Set up a very basic network in class as manually as possible with the following goals:
+
+- Allow students to relate familiar day-to-day user interfaces (i.e. a desktop browser or a mobile app) to the physical infrastructure that carry our communications (i.e. an ethernet cable or a wireless radio)
+- Allow opportunity for students to tinker with command line tools, and help each other out to encourage an atmosphere of peer-to-peer learning, since we expect varying degrees of experience and comfort with typing into a shell from our audience (small set of simple commands with recoverable states)
+- Concretize abstract concepts discussed earlier such as _IP addresses_ and _centralize points of failure in hierarchical networks_
+- As we go through the steps in the activity, make a connection between what we are building to concepts that we have discussed before
+
+### Materials
+
+- An even number of Raspberry Pi 3s
+- SD cards with mesh-orange images with all necessary interfaces pre-configured and programs pre-installed (e.g. ifconfig, netcat, cjdns)
+- Ethernet cables equaling the number of Raspberry Pis
+- Laptops that can make a SSH session to each Raspberry Pi (each running a Host AP with unique SSID)
```

Comment:
One issue we may have is as all these nodes come up there will be so many SSIDs with default passwords floating around that it may be difficult to identify which one is yours.

We will need a way to "lock in" the ssid for each node and record it with a label on the box.

Without using configs perhaps we can tie the ssid to the MAC address of the onboard wireless adapter? or maybe even Ethernet?

Reply:
Yes I think we must drop a unique ssid in `conf.d` for each node. I already have to do this for two nodes otherwise I don't know which **test2** I am connected to.